### PR TITLE
:cherries: :bug: Try to fit annotations into on-demand imports for member che…

### PR DESCRIPTION
**Pull request by jmle**
…cking (#208)

* Try to fit annotations into on-demand imports for member checking
* Remove unneeded document
* Address automatic PR review comments

:warning: DNM until release-0.9 is cut

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced annotation resolution to evaluate multiple fully-qualified name candidates for more accurate matching.
  * Added support for nested annotations within member-value pairs so inner annotations are properly recognized.
  * Improved import handling to better resolve on-demand and single-type annotation names when determining candidates.

### CODE REVIEW

**Comment by shawn-hurley** [comment]
If this is a cherry-pick, should we be release-0.9?

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java
Comment (shawn-hurley) [inline java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java:195]: super silly question but could I import xxx.yy.zzz.* that as ccc.AAA annotation and just reference it as (@ccc.AAA)? 

If that makes no sense I get it, just wondering if that is a usecase that the java language supports that we have to handle.

**Review by shawn-hurley — APPROVED** [review]
Just wondering about an edge case, generally I think this is better behavior.

Is there any way to add a test for this in the integration tests?

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java
Comment (jmle) [reply java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java:195]: Sorry mate I don't understand, do you mean using something like an `as` keyword? That doesn't exist in Java, but you can `import a.b.c.D;` and then also use `@a.b.c.D` in the code (although it wouldn't make much sense)

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java [hunk 1]
```diff
@@ -178,21 +184,37 @@ private IAnnotation[] tryToGetAnnotations(SourceRefElement t) {
...
+            return List.of(name);
+        }
+        List<IImportDeclaration> imports = tryToGetImports(annotation);
+        List<String> candidates = new ArrayList<>();
+        // Single-type: at most one
+        imports.stream()
+                .filter(i -> i.getElementName().endsWith(name))
+                .findFirst()
+                .map(IImportDeclaration::getElementName)
+                .ifPresent(candidates::add);
```
[the remark this answers is not in the document]
Comment (jmle) [reply java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java:206]: I have addressed this, please take a look.

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java [hunk 2]
```diff
@@ -47,28 +48,33 @@ default boolean matchesAnnotationQuery(SearchMatch match, List<Class<? extends S
...
-                                if (member.getValue() instanceof Object[]) {
-                                    Object[] objs = (Object[]) member.getValue();
-                                    for (int i = 0; i < objs.length; i++) {
-                                        Annotation innerAnnotation = (Annotation) objs[i];
-                                        fqn = getFQN(innerAnnotation);
+                    for (String fqn : getFQNCandidates(annotation)) {
+                        if (getAnnotationQuery().matchesAnnotation(fqn)) {
+                            return doElementsMatch((Annotation) annotation);
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (jmle) [reply java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java:55]: I have addressed this, please take a look.

File: java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java [hunk 3]
```diff
@@ -178,21 +184,37 @@ private IAnnotation[] tryToGetAnnotations(SourceRefElement t) {
...
-                    .map(IImportDeclaration::getElementName)
-                    .orElse("");
+        if (name == null || name.isEmpty()) {
+            return List.of();
+        }
+        // Already fully qualified in source (e.g. @com.example.MyAnnotation)
+        if (name.contains(".")) {
+            return List.of(name);
+        }
+        List<IImportDeclaration> imports = tryToGetImports(annotation);
```
[the remark this answers is not in the document]
Comment (jmle) [reply java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/WithAnnotationQuery.java:199]: We can leave this for another PR.

**Comment by jmle** [comment]
> If this is a cherry-pick, should we be release-0.9?

Yep totally
